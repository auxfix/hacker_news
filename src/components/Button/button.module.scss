@import '~styles/common.scss';

.container {
   display: inline-block;
   padding: 1.5rem 3rem;
   letter-spacing: 1px;
   text-transform: uppercase;
   font-size: 1.2rem;
   color: $white;
   border-radius: .8rem;
   text-decoration: none;
   background-image: linear-gradient(to right, $blue_gradient_start 0%, $blue_gradient_end 100%);
   &:hover{
       background-image: linear-gradient(to right, $blue_gradient_end 0%, $blue_gradient_start 100%);
   }
}