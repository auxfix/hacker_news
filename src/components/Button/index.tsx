import styles from './button.module.scss';
import useButtonStyle from './hooks/useButtonStyle';

interface ButtonProps {
  children: React.ReactNode;
  onClick: () => void;
}

export default function Button(props: ButtonProps) {
  const { children, onClick } = props;
  const [buttonStyle, onButtonDown, onButtonUp] = useButtonStyle([
    styles.button,
    styles.buttononpush,
  ]);

  return (
    <button
      className={buttonStyle}
      onClick={onClick}
      onMouseDown={onButtonDown}
      onMouseUp={onButtonUp}
      type="button"
      data-testid="button"
    >
      {children}
    </button>
  );
}
