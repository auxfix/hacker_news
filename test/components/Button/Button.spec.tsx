import React from 'react';
import Button from 'components/Button';
import { fireEvent, render, screen } from 'test-utils';

const mockOnButtonClick = jest.fn();

describe('Button', () => {
  it('should render properly', () => {
    const buttonText = 'Click me';

    render(<Button onClick={mockOnButtonClick}>{buttonText}</Button>);

    expect(screen.getByRole('button')).toMatchSnapshot();
    expect(screen.getByRole('button')).toHaveTextContent(buttonText);
  });

  it('on click handler is working', () => {
    const buttonText = 'Click me';

    render(<Button onClick={mockOnButtonClick}>{buttonText}</Button>);

    fireEvent.click(screen.getByRole('button'));

    expect(mockOnButtonClick).toHaveBeenCalledTimes(1);
  });
});
