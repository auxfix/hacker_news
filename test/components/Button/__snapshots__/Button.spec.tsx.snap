// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
<button
  data-testid="button"
  type="button"
>
  Click me
</button>
`;
